docs/spec_draft.md

```diff
@@ -1263,6 +1265,116 @@ Numeric precision is implementation-defined.
+  perform a less-than comparison. Formally, after the array is sorted, it holds
+  for all index positions `i, j` with `i < j` that either
+  `comparator(value[i], value[j]) = comparator(value[j], value[i]) = false` or
+  `comparator(value[i], value[j]) = true`.
+
+* If the operand has higher rank, the operand is sorted along the provided
+  dimension. For example, for a rank-2 tensor (a matrix), a dimension value of
+  `0` will independently sort every column, and a dimension value of `1` will
+  independently sort each row. If no dimension number is provided, then the last
+  dimension is chosen by default. For the dimension which is sorted, the same
+  sorting order applies as in the rank-1 case.
+
+If `n > 1` operands are provided:
+
+* All `n` operands must be tensors with the same dimensions. The element types
+  of the tensors may be different.
+
+* All operands are sorted together, not individually. Conceptually the operands
+  are treated as a tuple. When checking whether the elements of each operand at
+  index positions `i` and `j` need to be swapped, the comparator is called with
+  `2 * n` scalar parameters, where parameter `2 * k` corresponds to the value at
+  position `i` from the `k-th` operand, and parameter `2 * k + 1` corresponds to
+  the value at position `j` from the `k-th` operand. Usually, the comparator
+  would thus compare parameters `2 * k` and `2 * k + 1` with each other and
+  possibly use other parameter pairs as tie breakers.
+
+* The result is a tuple that consists of the operands in sorted order (along
+  the provided dimension, as above). The `i-th` operand of the tuple corresponds
+  to the `i-th` operand of Sort.
+
+### Operands
+
+| Name        | Type                                              |
+|-------------|---------------------------------------------------|
+| `inputs`    | variadic number of tensors of any supported types |
+| `dimension` | `si64` (defaults to `-1` or `R-1`)                |
+| `is_stable` | `i1`   (defaults to `false`)                      |
+
+### Regions
+
+| Name         | Type               |
+|--------------|--------------------|
+| `comparator` | a region of size 1 |
+
+### Results
+
+| Name     | Type                                              |
+|----------|---------------------------------------------------|
+| `result` | variadic number of tensors of any supported types |
+
+### Constraints
+
+  * (C1) `inputs` should have alteast 1 tensor.
+  * (C2) All tensors in `input` have same shape, they might have different
+         element types.
+  * (C3) 0 $\le$ `dimension` $\lt$ rank of `inputs[0]`.
+  * (C4) `comparator` takes arguments equal to twice the number of tensors in
+         `inputs`.
+  * (C5) Type of `2*i`th and `2*i+1`th parameter of `comparator` funcion is
+         same/compatible with element type of `i`th tensor in `input`.
+  * (C6) Result of `comparator` is a single 0-ranked tensor with element-type
+         i1.
```

Comment:
Let's make (C4)-(C6) more concise, e.g. collapse them into one constraint and rewrite them along the lines of "`comparator` has type `(E1, E1, ..., EN-1, EN-1) -> i1`, where `Ei` is the element type of `inputs[i]`".